pallets/liquidity-proxy/src/lib.rs

```diff
@@ -544,18 +546,13 @@ impl<T: Config> Pallet<T> {
                 .into_iter()
                 .filter(|(_src, part_amount)| part_amount.amount() > balance!(0))
                 .map(|(src, part_amount)| {
-                    let part_amount = part_amount.amount();
-                    let part_limit = (FixedWrapper::from(part_amount) / amount.amount()
-                        * amount.limit())
-                    .try_into_balance()
-                    .map_err(|_| Error::CalculationError::<T>)?;
```

Comment:
This is the root cause of the problem. `part_limit` was calculated as a proportion of original `amount`.
Now ALT works with different liquidity sources, which can have very difference prices, and as a result - different limit parts. Just "proportion" doesn't work anymore.
New approach is the calculation the limit part for each liquidity source separately and provide id in `AggregatedSwapOutcome.distribution` (see liquidity_aggregator.rs).